Add documentation for users to run e2e tests locally (#449)
arlonproj/arlon: docs/e2e_testing.md

- Currently, we are running the arlon e2e tests on an ubuntu VM using Github Actions.

- To invoke the arlon e2e tests, we are adding a new build target `make test-e2e` This command will execute the [e2e test job](https://github.com/arlonproj/arlon/blob/main/.github/workflows/e2e.yaml) using Github Actions.

### Running Arlon Tests Locally

- To run the arlon e2e tests locally on your developer setup (Linux or MacOS), export the following environment variables that are required to download CAPA and to setup the git server.
    - GIT_USER
    - GIT_PASSWORD
    - GIT_EMAIL
    - AWS_ACCESS_KEY_ID
    - AWS_SECRET_ACCESS_KEY
    - AWS_REGION
    - AWS_SSH_KEY_NAME
    - AWS_CONTROL_PLANE_MACHINE_TYPE
    - AWS_NODE_MACHINE_TYPE

> - After exporting these variables, run the make target `make test-e2e` from the top of the repository path.

Reviewer: ```suggestion
- After exporting these variables, run the make target `make test-e2e` from the repository root.
```

- Currently, we are running the arlon e2e tests on an ubuntu VM using Github Actions.

> - To invoke the arlon e2e tests, we are adding a new build target `make test-e2e` This command will execute the [e2e test job](https://github.com/arlonproj/arlon/blob/main/.github/workflows/e2e.yaml) using Github Actions.

Reviewer: ```suggestion
- To invoke the arlon e2e tests, we are adding a new build target `make test-e2e` This command will be executed the [e2e test job](https://github.com/arlonproj/arlon/blob/main/.github/workflows/e2e.yaml) using Github Actions.
```

> - To invoke the arlon e2e tests, we are adding a new build target `make test-e2e` This command will execute the [e2e test job](https://github.com/arlonproj/arlon/blob/main/.github/workflows/e2e.yaml) using Github Actions.

Reviewer: ```suggestion
- To invoke the arlon e2e tests, we are adding a new build target `make test-e2e` This command will be executed the [e2e test job](https://github.com/arlonproj/arlon/blob/main/.github/workflows/e2e.yaml) using Github Actions.
```

### Running Arlon Tests Locally

- To run the arlon e2e tests locally on your developer setup (Linux or MacOS), export the following environment variables that are required to download CAPA and to setup the git server.

>     - GIT_USER - Github username
>     - GIT_PASSWORD - Github password
>     - GIT_EMAIL - Github email

Reviewer: ```suggestion
    - GIT_USER - Username used by gitea helm chart for setting up test user
    - GIT_PASSWORD - Password for the test user
    - GIT_EMAIL - Dummy email value for the test user
```

    - AWS_ACCESS_KEY_ID - AWS access key
    - AWS_SECRET_ACCESS_KEY - AWS secret access key
    - AWS_REGION - AWS region in which you want to run these tests

>     - AWS_SSH_KEY_NAME -AWS SSH KEY in the AWS REGION where you want to run these tests

Reviewer: ```suggestion
    - AWS_SSH_KEY_NAME - refers to the AWS SSH key name in the region specified by `AWS_REGION`
```

    - AWS_CONTROL_PLANE_MACHINE_TYPE is the instance type of the control plane machines( e.g. t2.medium, t3.large)
    - AWS_NODE_MACHINE_TYPE is the instance type of the worker node machines( e.g. t2.medium, t3.large)

- After exporting these variables, run the make target `make test-e2e` from the repository root.

> - This will run the arlon e2e tests and clean up all the resources created by these tests post the completion of this test. Incase, the teardown is not triggered via this target, manually run the teardown script: `./testing/e2e_setup_teardown.sh`.

Reviewer: we have a make target for teardown, we should mention that

> - The cleanup of AWS resources is a little unreliable and it is recommended to clean all the AWS resources manually just to be completely sure about the deletion.

Reviewer: ```suggestion
- The cleanup of AWS resources is a unreliable due to bugs in CAPA, and it is advised to clean all the AWS resources manually to avoid incurring costs. In particular, any one or more of these: the VPC, instances for the nodes and the control plane were found to be left out in out tests.
```
